includes/class-job-listing-stats.php

```diff
@@ -38,42 +38,21 @@ class Job_Listing_Stats {
 	/**
 	 * Stats for a single job listing.
 	 *
-	 * @param int $job_id
+	 * @since $$next-version$$
+	 *
+	 * @param int  $job_id
+	 * @param null $date_range
 	 */
-	public function __construct( $job_id ) {
+	public function __construct( $job_id, $date_range = null ) {
 
 		$this->job_id     = $job_id;
-		$this->start_date = get_post_datetime( $job_id )->format( 'Y-m-d' );
-
-	}
+		$this->start_date = $date_range[0] ?? get_post_datetime( $job_id )->format( 'Y-m-d' );
+		$this->end_date   = $date_range[1] ?? ( new \DateTime( 'today' ) )->format( 'Y-m-d' );
```

Comment:
Both of these should probably use get_option( 'date_format' ) ?: 'F j, Y' as in #2779 ?

Reply:
This one is just for the array keys, matching the YYYY-MM-DD date format from the DB, not user-visible.